### PR TITLE
[ShaderGraph] Fix null exception when subgraph conversion overwrites itself

### PR DESCRIPTION
**Pull request by esmelusina**
### Purpose of this PR
Quick PR to address this bug: https://fogbugz.unity3d.com/f/cases/1268639/. This fix just prevents the offending null exception and provides a friendly prompt to warn that you will be overwriting the file you are currently working on.

### Testing status

**Manual Tests**: What did you do?
Create new subgraph with a single node output to the subgraph's output.
Select node -> convert to -> Sub-Graph.
Save As same file name as existing subgraph.
Reload when prompted.

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

### CODE REVIEW

File: com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs [hunk 1]
```diff
@@ -472,6 +472,16 @@ public void ToSubGraph()
 
             path = EditorUtility.SaveFilePanelInProject("Save Sub Graph", "New Shader Sub Graph", ShaderSubGraphImporter.Extension, "", path);
             path = path.Replace(Application.dataPath, "Assets");
+
+            // Friendly warning that the user is generating a subgraph that would overwrite the one they are currently working on.
+            if (AssetDatabase.AssetPathToGUID(path) == selectedGuid)
+            {             
+                if (!EditorUtility.DisplayDialog("Overwrite Current Subgraph", "You are currently working on this Subgraph, are you sure you want to overwrite?", "Yes", "Cancel"))
```
Comment (alindmanUnity) [inline com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs:479]: Do we want to also explicitly state that overwriting the current subgraph will destroy data?

Comment (esmelusina) [reply com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs:479]: The 'Save As' provides an overwrite prompt already. I think the second one here is probably enough to make the point. If you have specific language you think would be best I'm happy to change it to use that. I do think 'destroy data' is a rich concept with regards to language and context, whereas 'overwrite' and 'cannot be undone' is commonly understood enough to capture that idea.

I'll go ahead and add a 'cannot be undone' disclaimer.

Comment (esmelusina) [reply com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs]: Sharlene gave me some much better language to use.

**Comment by alindmanUnity** [comment]
hdrp failures consistent with master, merging

**Pull request merged**